application/helpers/frontend_helper.php

```diff
@@ -1886,13 +1889,39 @@ function doAssessment($surveyid, $returndataonly=false)
                             if ($usresult)
                             {
                                 $usrow = $usresult->read();
-                                $fieldmap[$field['fieldname']]['assessment_value']=$usrow['assessment_value'];
-                                $total=$total+$usrow['assessment_value'];
+                                $assessmentValue = $usrow['assessment_value'];
                             }
                         }
                     }
                     $groups[]=$field['gid'];
                 }
+
+                // If this is a question (and not a survey field, like ID), save asessment value
+                if (!empty($field['title']))
+                {
+                    /**
+                     * Allow Plugin to update assessment value
+                     */
+                    // Prepare Event Info
+                    $event = new PluginEvent('afterSurveyQuestionAssessment');
+                    $event->set('surveyId', $surveyid);
+                    $event->set('question', $field);
```

Comment:
> It's the first plugin who send information for Sub question (the final answer X times for array/multiple choice XxY times for array text/number : if we send ABC information in this plugin, we must send ABC information in next plugin too ....

Partly agree. I am more into saying some basic information shall be shared among plugins and that there can be specific info for each plugin.

> For the question text: get it for model.
You mean, get it from the Model (from DB), right?

>  But micro memory + micro memory + .... make 512 minimum to lauch a little survey.
I understand... I agree for the question text. Still, sending the question title (limited string) could be a good idea, as it is much more basic info. Don't you agree?

Thinking forward, maybe it is just easier to use a singleton to store fieldmap (or similar) as a app-wide cache.